pkg/nodes/merge/merge.go

```
@@ -14,16 +18,112 @@
+
+	var resultData interface{}
+
+	switch first.(type) {
+	case map[string]interface{}:
+		switch strategy {
+		case "deep":
+			merged := make(map[string]interface{})
+			for _, item := range dataSlice {
+				if m, ok := item.(map[string]interface{}); ok {
+					// mergo.Merge performs deep merge with override
+					mergo.Merge(&merged, m, mergo.WithOverride, mergo.WithAppendSlice)
```

golangci-lint (1.64.8)

Error return value of `mergo.Merge` is not checked

(errcheck)